accumulate: test failure message prints unhelpful function pointer, go vet complains

I know there was a past work on this in https://github.com/exercism/exercism.io/pull/1260 , but now `go vet` has something to say again:

```
accumulate_test.go:33: arg test.converter in printf call is a function value, not a function call
```

Here is an example of the error message you will see if your implementation fails a test:

```
--- FAIL: TestAccumulate (0.00s)
	accumulate_test.go:35: PASS: echo []
	accumulate_test.go:35: PASS: echo [echo echo echo echo]
	accumulate_test.go:33: Accumulate([first letter only], (func(string) string)(0x47de00)): expected [First Letter Only], actual [first letter only]
```

That `(func(string) string)(0x47de00)` is decidedly unhelpful. I can think of two possibilities to replace:

1. The name of the function - however that appears to need reflection, unless I am mistaken.
2. The description (just like the PASS message does)

--- a/exercises/accumulate/accumulate_test.go
+++ b/exercises/accumulate/accumulate_test.go
@@ -23,14 +23,14 @@ var tests = []struct {
 	{[]string{}, []string{}, echo, "echo"},
 	{[]string{"echo", "echo", "echo", "echo"}, []string{"echo", "echo", "echo", "echo"}, echo, "echo"},
 	{[]string{"First", "Letter", "Only"}, []string{"first", "letter", "only"}, capitalize, "capitalize"},
-	{[]string{"HELLO", "WORLD"}, []string{"hello", "world"}, strings.ToUpper, "upcase"},
+	{[]string{"HELLO", "WORLD"}, []string{"hello", "world"}, strings.ToUpper, "strings.ToUpper"},
 }
 
 func TestAccumulate(t *testing.T) {
 	for _, test := range tests {
 		actual := Accumulate(test.given, test.converter)
 		if fmt.Sprintf("%s", actual) != fmt.Sprintf("%s", test.expected) {
-			t.Fatalf("Accumulate(%s, %#v): expected %s, actual %s", test.given, test.converter, test.expected, actual)
+			t.Fatalf("Accumulate(%s, %s): expected %s, actual %s", test.given, test.description, test.expected, actual)
 		} else {
 			t.Logf("PASS: %s %v", test.description, test.given)
 		}

